addons/l10n_in_hr_holidays/models/hr_leave.py

```diff
@@ -0,0 +1,76 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime, timedelta
+
+from odoo import models, fields
+
+
+class HolidaysRequest(models.Model):
+    _inherit = "hr.leave"
+
+    l10n_in_contains_sandwich_leaves = fields.Boolean()
+
+    def _l10n_in_apply_sandwich_rule(self, public_holidays, employee_leaves):
+        if not self.request_date_from or not self.request_date_to:
+            return
+        date_from = self.request_date_from
+        date_to = self.request_date_to
+        total_leaves = (self.request_date_to - self.request_date_from).days + 1
+
+        def is_non_working_day(date):
+            return date.weekday() >= 5 or any(
+                datetime.date(holiday['date_from']) <= date <= datetime.date(holiday['date_to']) for holiday in public_holidays
+            )
```

Comment:
this depends on the calendar of the employee as well as the schedule of the company so this is wrong.
Maybe move the method `_works_on_date` defined in the `l10n_fr_hr_holidays` module into the `resource` module so that you can use that method